localstack/services/generic_proxy.py

```diff
@@ -186,6 +193,130 @@ def return_response(
         return None
 
 
+class PartitionAdjustingProxyListener(MessageModifyingProxyListener):
```

Comment:
similarly i think the name paritionadjusting is not very clear. what about something more specific like `ArnPartitionRewriteListener`? naming is hard :grin:

Reply:
Renamed to `ArnPartitionRewriteListener`.